frontend/public/components/instantiate-template.tsx

```diff
@@ -136,16 +136,20 @@ const TemplateForm_: React.FC<TemplateFormProps> = (props) => {
   const [parameters, setParameters] = React.useState([]);
   const [inProgress, setInProgress] = React.useState(false);
   const [error, setError] = React.useState('');
+  const isInitialLoad = React.useRef(true);
 
   const { t } = useTranslation();
   const navigate = useNavigate();
 
   React.useEffect(() => {
-    const object = (obj.data.parameters || []).reduce((acc, { name, value }) => {
-      acc[name] = value;
-      return acc;
-    }, {});
-    setParameters(object);
+    if (isInitialLoad.current && obj.loaded) {
```

Comment:
Hi @vikram-raj ,

Issue is not with the [Firehose](https://github.com/openshift/console/blob/master/frontend/public/components/instantiate-template.tsx#L362), it is the way this form is implemented. I tried with `useK8sGet` but this issue is still there. 

Issue is [here](https://github.com/openshift/console/blob/master/frontend/public/components/instantiate-template.tsx#L283), as soon as user enters something, useK8sGet is also getting called and if parameter value is not there it is setting as null and if parameter is defined also if user changes something, that will be replaced.

I think solution provided is a better one because it is not required to set any value once we come into this form and after the first load, so it will restrict auto filling of the parameters from the second time.